Formula/fn.rb

```diff
@@ -0,0 +1,52 @@
+class Fn < Formula
+  desc "Command-line tool for the fn project"
+  homepage "https://fnproject.github.io"
+  url "https://github.com/fnproject/cli/archive/0.3.84.tar.gz"
+  sha256 "495c93a955b2a4cfb0a2ea4a32cd150fc2a98192051d99a51a1f63bddfe178fc"
+
+  depends_on "go" => :build
+  depends_on "glide" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GLIDE_HOME"] = HOMEBREW_CACHE/"glide_home/#{name}"
+    dir = buildpath/"src/github.com/fnproject/cli"
+    dir.install Dir["*"]
+    cd dir do
+      system "glide", "install", "-v", "--force", "--skip-test"
+      system "go", "build", "-o", "#{bin}/fn"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/fn --version")
+    system "#{bin}/fn", "init", "--runtime", "go", "--name", "myfunc"
+    assert_predicate testpath/"func.go", :exist?, "expected file func.go doesn't exist"
+    assert_predicate testpath/"func.yaml", :exist?, "expected file func.yaml doesn't exist"
+    require "socket"
+    server = TCPServer.new("localhost", 0)
+    port = server.addr[1]
+    pid = fork do
+      loop do
+        socket = server.accept
+        response = '{"route": {"path": "/myfunc", "image": "fnproject/myfunc"} }'
+        socket.print "HTTP/1.1 200 OK\r\n" \
+                    "Content-Length: #{response.bytesize}\r\n" \
+                    "Connection: close\r\n"
+        socket.print "\r\n"
+        socket.print response
+        socket.close
+      end
+    end
+    begin
+      expected = "/myfunc created with fnproject/myfunc"
+      # output = shell_output("API_URL=http://localhost:#{port} #{bin}/fn deploy --app myapp --local --registry funcy")
+      output = shell_output("API_URL=http://localhost:#{port} FN_REGISTRY=fnproject #{bin}/fn routes create myapp myfunc")
```

Comment:
Any chance `API_URL` and `FN_REGISTRY` can be set using `ENV[""]` rather than being whacked before the command like this? Maybe not, but, harmless to ask. Lengthy lines are a bit grim.

Reply:
Done.